Failed to build on FreeBSD

```
$ cmake -S . -B build
-- The CXX compiler identification is Clang 18.1.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- -------------YLT INSTALL SETTING------------
-- INSTALL_THIRDPARTY: ON
-- INSTALL_STANDALONE: ON
-- INSTALL_INDEPENDENT_THIRDPARTY: ON
-- INSTALL_INDEPENDENT_STANDALONE: ON
-- --------------------------------------------
-- -------------YLT COMPILE SETTING------------
-- CXX Standard: 20
-- Build type: Release
-- BUILD_WITH_LIBCXX: OFF
-- ENDIAN: LITTLE
-- USE_CCACHE: OFF
-- --------------------------------------------
-- -------------YLT DEVELOP SETTING------------
-- BUILD_EXAMPLES: ON
-- BUILD_BENCHMARK: ON
-- BUILD_UNIT_TESTS: ON
-- COVERAGE_TEST: OFF
-- GENERATE_BENCHMARK_DATA: ON
-- CORO_RPC_USE_OTHER_RPC: OFF
-- --------------------------------------------
-- -------------YLT CONFIG SETTING-------------
-- YLT_ENABLE_SSL: OFF
-- YLT_ENABLE_PMR: OFF
-- YLT_ENABLE_IO_URING: OFF
-- YLT_ENABLE_STRUCT_PACK_UNPORTABLE_TYPE: OFF
-- YLT_ENABLE_STRUCT_PACK_OPTIMIZE: OFF
-- --------------------------------------------
-- -------------YLT PROJECT SETTING------------
-- BUILD_CORO_HTTP: ON
-- BUILD_CORO_IO: ON
-- BUILD_CORO_RPC: ON
-- BUILD_EASYLOG: ON
-- BUILD_INCLUDE: ON
-- BUILD_METRIC: ON
-- BUILD_REFLECTION: ON
-- BUILD_STRUCT_JSON: ON
-- BUILD_STRUCT_PACK: ON
-- Protobuf_FOUND: TRUE
-- BUILD_STRUCT_PB: ON
-- BUILD_STRUCT_XML: ON
-- BUILD_STRUCT_YAML: ON
-- BUILD_UTIL: ON
-- --------------------------------------------
-- Configuring done (2.6s)
-- Generating done (0.2s)
-- Build files have been written to: /work/yalantinglibs/build
```

```
$ gmake -C build
gmake: Entering directory '/work/yalantinglibs/build'
[  1%] Building CXX object src/coro_http/examples/CMakeFiles/coro_http_example.dir/example.cpp.o
In file included from /work/yalantinglibs/src/coro_http/examples/example.cpp:18:
In file included from /work/yalantinglibs/include/ylt/coro_http/coro_http_client.hpp:20:
In file included from /work/yalantinglibs/include/ylt/easylog.hpp:33:
In file included from /work/yalantinglibs/include/ylt/easylog/appender.hpp:27:
/work/yalantinglibs/include/ylt/easylog/record.hpp:242:13: error: use of undeclared identifier 'SYS_thr_self'
  242 |     syscall(SYS_thr_self, &tid);
      |             ^
1 error generated.
gmake[2]: *** [src/coro_http/examples/CMakeFiles/coro_http_example.dir/build.make:76: src/coro_http/examples/CMakeFiles/coro_http_example.dir/example.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:485: src/coro_http/examples/CMakeFiles/coro_http_example.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
gmake: Leaving directory '/work/yalantinglibs/build'
```

--- a/include/ylt/easylog/record.hpp
+++ b/include/ylt/easylog/record.hpp
@@ -21,11 +21,15 @@
 
 #include "ylt/util/time_util.h"
 #ifdef __linux__
-#include <sys/syscall.h>
 #include <unistd.h>
 #endif
 #if __has_include(<memory_resource>)
 #include <memory_resource>
+#endif
+#if defined(__linux__) || defined(__FreeBSD__)
+#include <sys/syscall.h>
+#elif defined(__rtems__)
+#include <rtems.h>
 #endif
 #include <sstream>
 #include <string>

